tests/testsuite/update.rs

```diff
@@ -1740,34 +1772,113 @@ fn update_breaking() {
         .publish();
 
-    p.cargo("update --dry-run --breaking")
-        .with_status(1)
-        .with_stderr_contains("[ERROR] unexpected argument '--breaking' found")
+    p.cargo("update -Zunstable-options --dry-run --breaking")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        // .env("CARGO_LOG", "cargo::ops=trace")
+        .with_stderr_unordered(
+            "\
+[UPDATING] `alternative` index
+[UPGRADING] alternative-1 ^1.0 -> v2.0.0
+[UPGRADING] alternative-2 ^1.0 -> v2.0.0
+[UPDATING] `dummy-registry` index
+[UPGRADING] incompatible ^1.0 -> v2.0.0
+[UPGRADING] multiple ^1.0 -> v3.0.0
+[UPGRADING] ws ^1.0 -> v2.0.0
+[UPGRADING] multiple ^2.0 -> v3.0.0
+[LOCKING] 5 packages to latest compatible versions
+[UPDATING] alternative-1 v1.0.0 (registry `alternative`) -> v2.0.0
+[UPDATING] alternative-2 v1.0.0 (registry `alternative`) -> v2.0.0
+[UPDATING] incompatible v1.0.0 -> v2.0.0
+[REMOVING] multiple v1.0.0
+[REMOVING] multiple v2.0.0
+[ADDING] multiple v3.0.0
+[UPDATING] ws v1.0.0 -> v2.0.0
+[NOTE] pass `--verbose` to see 4 unchanged dependencies behind latest
+[WARNING] not updating lockfile due to dry run
+[WARNING] not upgrading manifest `[..]/Cargo.toml` due to dry run
+[WARNING] not upgrading manifest `[..]/foo/Cargo.toml` due to dry run
+[WARNING] not upgrading manifest `[..]/bar/Cargo.toml` due to dry run
+",
+        )
         .run();
 
-    p.cargo("update --breaking")
-        .with_status(1)
-        .with_stderr_contains("[ERROR] unexpected argument '--breaking' found")
+    p.cargo("update -Zunstable-options --breaking")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        .with_stderr_unordered(
+            "\
+[UPDATING] `alternative` index
+[UPGRADING] alternative-1 ^1.0 -> v2.0.0
+[UPGRADING] alternative-2 ^1.0 -> v2.0.0
+[UPDATING] `dummy-registry` index
+[UPGRADING] incompatible ^1.0 -> v2.0.0
+[UPGRADING] multiple ^1.0 -> v3.0.0
+[UPGRADING] ws ^1.0 -> v2.0.0
+[UPGRADING] multiple ^2.0 -> v3.0.0
+[LOCKING] 5 packages to latest compatible versions
+[UPDATING] alternative-1 v1.0.0 (registry `alternative`) -> v2.0.0
+[UPDATING] alternative-2 v1.0.0 (registry `alternative`) -> v2.0.0
+[UPDATING] incompatible v1.0.0 -> v2.0.0
+[REMOVING] multiple v1.0.0
+[REMOVING] multiple v2.0.0
+[ADDING] multiple v3.0.0
+[UPDATING] ws v1.0.0 -> v2.0.0
+[NOTE] pass `--verbose` to see 4 unchanged dependencies behind latest
+",
+        )
         .run();
 
+    let root_manifest = p.read_file("Cargo.toml");
+    snapbox::assert_data_eq!(
+        &root_manifest,
+        r#"
+                # Check if formatting is preserved
+
+                [workspace]
+                members  =  ["foo", "bar"]
+
+                [workspace.dependencies]
+                ws  = "2.0"
+            "#,
+    );
+
+    let crate_manifest = p.read_file("foo/Cargo.toml");
+
+    // FIXME: Not sure how to preserve all formatting here.
+    snapbox::assert_data_eq!(
+        &crate_manifest,
+        r#"
```

Comment:
This should be created with `cargo_test_support::str![]` which will give us snapshot updating capabilities

Reply:
Fixed.